source/MRViewer/MRGLStaticHolder.cpp

```diff
@@ -1243,46 +1243,36 @@ void main(void)
   uniform vec4 color;
   uniform vec2 shift;
   uniform float blurRadius;
+  uniform bool convX;
   out vec4 outColor;                 // (out to render) fragment color
 
-  #define NUM_DIRECTIONS 12
-  #define QUALITY 3
-  
-  #define DIR_STEP 0.5235987756
-  #define INV_QUALITY 0.33333334
-  #define INV_NUM_SAMPLES 0.020833333
+  const float gaussWeights[] = float[6] (0.923, 0.726, 0.4867, 0.278, 0.135, 0.056);
+  const float invNorm = 0.161046;
 
   void main()
   { 
+    gl_FragDepth = 0.9999;
     ivec2 texSize = textureSize( pixels, 0 );
-    vec2 pos = gl_FragCoord.xy + shift;
+    vec2 pos = gl_FragCoord.xy;
+    if ( !convX )
```

Comment:
why?

Reply:
we use this shader for x convolution and for y convolution, shift can be applied only for one of them